packages/xod-client-browser/src/components/PopupInstallApp.jsx

```diff
@@ -34,12 +36,30 @@ class PopupInstallApp extends React.Component {
       <SkyLight
         hideOnOverlayClicked
         ref={this.assignPopupRef}
-        title="Oops! You need a desktop IDE!"
+        title="You need the desktop version of XOD IDE"
         afterClose={this.props.onClose}
       >
         <div className="ModalBody">
           <div className="ModalContent">
-            To upload projects you need to install XOD IDE for desktop.
+            Uploading a program requires access to USB ports. <br />
+            Browsers do not provide it for security reasons,&nbsp;
+            so you have to install more permissive IDE for the desktop.
+          </div>
+          <div className="ModalFooter">
+            <a
+              href="https://xod.io/downloads/?campaign=download&medium=upload_dialog"
```

Comment:
That’s not quite correct UTM args application. You need to use URL builder to get the correct one. See `packages/xod-client-browser/src/containers/App.jsx:202` for example.